CMakeLists.txt

```diff
@@ -16,24 +16,30 @@ set(GCF_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/git-cmake-format.py)
 
 find_package(ClangFormat ${GCF_CLANGFORMAT_MINIMAL_VERSION} REQUIRED)
 
-execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --absolute-git-dir
+execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --git-path hooks
   WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-  OUTPUT_VARIABLE GCF_GIT_ROOT
+  OUTPUT_VARIABLE GCF_GIT_HOOKS_DIR
   OUTPUT_STRIP_TRAILING_WHITESPACE)
-if(NOT GCF_GIT_ROOT)
+if(NOT GCF_GIT_HOOKS_DIR)
   message(WARNING "Not in a git repository")
 else()
-  set(GCF_GIT_DIR_SCRIPT ${GCF_GIT_ROOT}/hooks/git-cmake-format.py)
-  configure_file(
-    ${GCF_SCRIPT} ${GCF_GIT_DIR_SCRIPT}
-    COPYONLY)
-  unset(GCF_SCRIPT)
+  set(GCF_GIT_DIR_SCRIPT ${GCF_GIT_HOOKS_DIR}/git-cmake-format.py)
+  if((EXISTS ${GCF_GIT_DIR_SCRIPT}) AND (EXISTS ${GCF_GIT_HOOKS_DIR}/pre-commit))
+    message(STATUS "git-cmake-format.py already installed, skipping it")
+    else()
```

Comment:
nit: 
```suggestion
  else()
```

Reply:
In normal/worktree in works, and in submodule this does nothing, since `GCF_GIT_HOOKS_DIR` does not exist, so everythin is skipped.

Reply:
In practice, how do we compile the ginkgo in submodule?
I use `git submodule add <ginkgo_git>` run `git rev-parse --git-path hooks` in that folder.
I can get /home/mike/worktree/ginkgo/.git/modules/ginkgo/hooks
Is not it the GCF_GIT_HOOKS_DIR?

Reply:
Maybe I am wrong with the submodule process.
I do the following
```
git clone ginkgo 
cd ginkgo
mkdir dummy && cd dummy
git submodule add ginkgo
git checkout the branch and modify to use the commit
mkdir build && cd build
cmake -DGINKGO_DEVEL_TOOLS=ON ..
```
it gives /home/mike/worktree/ginkgo/.git/hooks (it needs to combine two path because I use the previous commit)

Reply:
The submodule issue is a bit more complex than expected. Due to the changes in 74de786 the hook and script are installed in the root repositories' `.git` if git-cmake-format is used in `add_subdirectory` (ie as a git submodule). I think this should be the expected behavior, since why would you include this repository if not to use the hook and script.

However, the issue becomes more complicated if ginkgo is used as a submodule. In that case the user would usually not set `GINKGO_DEVEL_TOOL=ON` and we don't have to worry about this (except for fixing the dummy hook). But what should happen if the user sets `GINKGO_DEVEL_TOOL=ON`? The options are:
1. write the hook+script into the users .git directory
2. skip/deactivate setting the ginkgo hooks
I would opt for 2., although I'm not sure how to figure out if ginkgo is used as a submodule.

Reply:
I personally would be fine with 1., since this setting is explicitly meant for developers working on Ginkgo only, and we don't need to make this too complicated.

Reply:
This could be a problem if the user already has a pre-commit hook, which we would overwrite. But I'm also fine with choosing 1. and figure out the overwrite problem if it ever appears.

Reply:
Maybe you can add a sentence or two to the INSTALL.md file detailing what we are doing (installing a pre-commit hook for formatting) more explicitly.

Reply:
I try to search the FETCH_CONTENT to check whether we can get who call the command.
but it seems not to have this variable.
We can add a GCF_BASE_DIR variable. if it is defined, use it. otherwise use CMAKE_SOURCE_DIR

I do not like 1 because it pollute ginkgo things to outside.
We can make sure ginkgo does not accidently add the hook to origin git if ginkgo is submodule first.

Reply:
you can use `git rev-parse --git-dir` (working dir needs to be CURRENT_SOURCE_DIR) to check the dir name is `.git` or not.